src/etc/inc/interfaces.inc

```diff
@@ -3272,18 +3272,23 @@ function interface_configure($interface = "wan", $reloadall = false, $linkupeven
 		interface_wireless_configure($realif, $wancfg, $wancfg['wireless']);
 	}
 
+	/* Get the vendor MAC.  Use source dependent upon whether or not booting. */
 	$current_mac = get_interface_mac($realhwif);
-	$vendor_mac = get_interface_vendor_mac($realhwif);
-	$spoof_mac = $wancfg['spoofmac'] ?: $vendor_mac;
+	if (platform_booting()) {
+		$vendor_mac = $current_mac;
+	} else {
+		$vendor_mac = get_interface_vendor_mac($realhwif);
+	}
```

Comment:
Interface configure function doesn't write the vendor MAC file.  Vendor MAC file is written only once, during boot by rc.bootup.  Same as before.  So don't see how it could end up with a spoofed mac being stored in the vendor_mac file.